File: 프로그래머스/lv0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.js
function solution(lines) {
    var answer = new Set();
    var all = new Set();
    lines.map((line) => {
        for(var i=line[0]; i<line[1]; i++){
            if(all.has(i)){
                answer.add(i);
            }
            all.add(i);
        }
    })
    return answer.size;
}

File: 프로그래머스/lv0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int solution(int[][] lines) {
        Set<Integer> answer = new HashSet<>();
        Set<Integer> set = new HashSet<>();
        for(int i=0; i<3; i++){
            for(int j=lines[i][0]; j<lines[i][1]; j++){
                if(!set.add(j)){
                    answer.add(j);
                }
            }
        }
        return answer.size();
    }
}